iceoryx_hoofs/include/iceoryx_hoofs/cxx/type_traits.hpp

```diff
@@ -45,6 +45,15 @@ struct add_const_conditionally<T, const C>
 template <typename T, typename C>
 using add_const_conditionally_t = typename add_const_conditionally<T, C>::type;
 
+///
+/// @brief Helper value to bind a static_assert to a type
+/// @code
+/// static_assert(always_false_v<Foo>, "Not implemented for the given type!");
+/// @endcode
+///
+template <typename>
+constexpr bool always_false_v = false;
```

Comment:
I remember I implemented something similar for the `cxx::string`. Maybe we can replace it there?

Reply:
Shall I create a new issue for this?

Reply:
Done in this PR